### PR TITLE
Get release notes from concourse/concourse

### PR DESCRIPTION
**Pull request by jomsie**
- concourse and release pipelines now get the release notes from concourse/release-notes
- moved other pipeline (prep-release-assets) from release notes into ci

Please review concourse/concourse#4312 first!

### CODE REVIEW

File: tasks/build-release-notes.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+platform: linux
+
+image_resource:
+  type: registry-image
+  source: {repository: concourse/unit}
+
+inputs:
+- name: concourse
+- name: version
+  optional: true
```
Comment (pivotal-jamie-klassen) [inline tasks/build-release-notes.yml]: you need another input `name: ci`

File: tasks/build-release-notes.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+outputs:
+- name: built-notes
+
+run:
+  path: scripts/build-release-notes
```
Comment (pivotal-jamie-klassen) [inline tasks/build-release-notes.yml]: this should be `path: ci/tasks/scripts/build-release-notes`

File: tasks/scripts/build-release-notes [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+elif [ "$#" -eq 1 ]; then
+  version=$1
+else
+  echo "version must be specified, either as an argument or an input" >&2
+  exit 1
+fi
+
+echo "$version" > ./built-notes/release-name
+# if this fails, it's because we forgot to manually promote latest.md to $version.md
+cp concourse/release-notes/"$version".md ./built-notes/notes.md
```
Comment (pivotal-jamie-klassen) [inline tasks/scripts/build-release-notes]: i think we can (and should) do this with an automated commit -- the constraint on whether a version is releasable should be controlled by `passed:` constraints in the pipeline, and something feels awkward to me about making a 'release commit' up front without first knowing whether the code in question is gonna pass... I can imagine some kind of deadlock where we think we can release, we promote the `latest.md` and then it turns out our code is broken, but by the time the next RC passes testing we've merged more features so our release notes have to change! and the cycle goes on.

However, the way it is here isn't gonna directly break anything, so let's save ^this work for another PR.

File: pipelines/release.yml [hunk 4]
```diff
@@ -1138,13 +1138,6 @@ resources:
     uri: https://github.com/concourse/docs
     branch: master
 
-- name: release-notes
-  type: git
-  icon: *git-icon
-  source:
-    uri: https://github.com/concourse/release-notes
-    branch: master
-
```
Comment (pivotal-jamie-klassen) [inline pipelines/release.yml]: :+1: very pleasing.

File: pipelines/concourse.yml [hunk 5]
```diff
@@ -1573,13 +1573,6 @@ resources:
     uri: https://github.com/concourse/docs
     branch: master
 
-- name: release-notes
-  type: git
-  icon: *git-icon
-  source:
-    uri: https://github.com/concourse/release-notes
-    branch: master
-
```
Comment (pivotal-jamie-klassen) [inline pipelines/concourse.yml]: 👍 very pleasing.

File: tasks/build-release-notes.yml (hunk 1 above)
Comment (deniseyu) [reply tasks/build-release-notes.yml]: d'oh. thanks for the catch

File: tasks/scripts/build-release-notes (hunk 3 above)
Comment (deniseyu) [reply tasks/scripts/build-release-notes]: yea, we were trying to come up with a process that encouraged a human to look over the contributor-submitted release notes as a whole and check that they're readable, but this workflow is clunky for sure. also tricky that we can't have the pipeline push directly to master and do this rename + push as part of the manually triggered release job from within the pipeline!

Comment (pivotal-jamie-klassen) [reply tasks/scripts/build-release-notes]: maybe the testing jobs could first check to see if the latest commit is only a documentation change, and automatically pass in that case...

**Pull request merged**